src/backend/backend.py

```diff
@@ -135,6 +135,11 @@ async def login_client(self,
             await client.close()
             raise
 
+        # Check if the user is already present on mirage
+        if client.user_id in self.clients.keys():
```

Comment:
You can do `if client.user_id in self.clients:`, no need to use `keys()`.
Looks good apart from that, thanks!